sys/saul_reg/saul_reg.c

```diff
@@ -126,3 +128,12 @@ int saul_reg_write(saul_reg_t *dev, phydat_t *data)
     }
     return dev->driver->write(dev->dev, data);
 }
+
+int saul_reg_test(saul_reg_t *dev)
+{
+
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+    return dev->driver->test(dev->dev);
```

Comment:
If you add a check for the NULL pointer `` if (dev->driver->test != NULL)`` here, then you do not need  the dummy function ``int saul_notsup(void *dev, phydat_t *dat)``